File: 参考/YCrispBiscuit-BestFriend/Server/app/controllers/health_controller.py
from fastapi import APIRouter, HTTPException
from app.schemas.health_schema import HealthResponse
from app.services.health_service import HealthService

router = APIRouter()
_service = HealthService()


@router.get("/health", response_model=HealthResponse)
async def health():
    try:
        """Route layer: 只负责 HTTP 层，调用 service 并返回 schema 定义的类型"""
        return await _service.get_health()
    except Exception:
        raise HTTPException(status_code=500, detail="Internal server error")

File: 参考/YanJing-Manager/RegionAI-Node-Server/YanJing-Node-Server/src/controllers/HealthController.ts
import { OpenAPIHono, createRoute } from '@hono/zod-openapi'
import { HealthService } from '../services/HealthService.js'
import { HealthResponseSchemaObj } from '../schemas/HealthSchema.js'
import { createResultSchema, Result, ErrorResultSchema } from '../utils/Result.js'

export const router = new OpenAPIHono()
const service = new HealthService()

const healthRoute = createRoute({
  method: 'get',
  path: '/',
  summary: '健康检查接口',
  description:
    '检测服务运行状态 (对应 FastAPI router.get("/health", response_model=HealthResponse))',
  tags: ['Health'],
  responses: {
    200: {
      description: '健康检查成功',
      content: {
        'application/json': {
          schema: createResultSchema(HealthResponseSchemaObj)
        }
      }
    },
    500: {
      description: '服务器内部错误',
      content: {
        'application/json': {
          schema: ErrorResultSchema
        }
      }
    }
  }
})

router.openapi(healthRoute, async (c) => {
  try {
    const data = await service.getHealth()
    return c.json(Result.success(data))
  } catch (e: unknown) {
    const message = e instanceof Error ? e.message : 'Internal Server Error'
    return c.json(Result.error(message, 500), 500)
  }
})
